system-tests/create_filewriter_image.py

```diff
@@ -0,0 +1,184 @@
+                os.environ["local_conan_server"]
+            )
+        )
+    else:
+        print("No local conan server available in environment variables.")
+    return container
+
+
+def kill_and_remove(container):
+    try:
+        container.kill()
+        container.remove()
+    except docker.errors.APIError as e:
+        pass
+
+
+def run_conan(container):
+    print("Running conan")
+    container.exec_run("mkdir build", workdir="/home/jenkins/")
+    execute_command(
+        "conan install --build=outdated ../conan", "/home/jenkins/build", container
+    )
+    execute_command("cp conan_hash_new.txt conan_hash.txt", "/home/jenkins/", container)
+    print("Done running conan")
+
+
+def rebuild_filewriter(container):
+    print("Re-building the filewriter")
+    execute_command(
+        'bash -c "bash activate_run.sh && cmake .. -GNinja -DCONAN=MANUAL-DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=False -DRUN_DOXYGEN=False && ninja kafka-to-nexus"',
+        "/home/jenkins/build",
+        container,
+    )
+    execute_command("cp src_hash_new.txt src_hash.txt", "/home/jenkins/", container)
+    print("Done building the filewriter")
+
+
+def re_generate_test_image(container):
+    print("Generating docker image")
+    try:
+        client.images.remove(TEST_IMAGE_NAME + ":latest")
+    except docker.errors.ImageNotFound as e:
+        pass
+    except docker.errors.APIError as e:
+        pass
+    container.commit(TEST_IMAGE_NAME, changes='CMD ["/home/jenkins/docker_launch.sh"]')
+
+
+def conan_hash_changed(container):
+    return (
+        container.exec_run("less conan_hash_new.txt").output
+        != container.exec_run("less conan_hash.txt").output
+    )
+
+
+def src_hash_changed(container):
+    return (
+        container.exec_run("less src_hash_new.txt").output
+        != container.exec_run("less src_hash.txt").output
+    )
+
+
+def create_filewriter_image():
+    list_of_containers = client.containers.list("all")
+    found_container = False
+    for c_container in list_of_containers:
+        if c_container.attrs["Name"][1:] == CONTAINER_NAME:
+            print("Found existing filewriter container")
+            found_container = True
+            container = client.containers.get(CONTAINER_NAME)
+            if container.attrs["Config"]["Image"] != IMAGE_NAME:
+                print("Filewriter container uses outdated image, ignoring")
+                kill_and_remove(container)
+                found_container = False
+            break
+    if not found_container:
+        print("Unable to find container, re-generating")
+        container = generate_new_container()
+    copy_to_container(container)
+    try:
+        container.start()
+    except docker.errors.APIError as e:
+        pass
+
+    if conan_hash_changed(container) or src_hash_changed(container):
+        container.exec_run('bash -c "rm -rf *"', workdir="/home/jenkins/build/")
+        run_conan(container)
```

Comment:
Does the conan bit need to be rerun if only the src hash has changed?

Reply:
In principle no. However, because the same (base) container is used for every system regardless of which PR, I am worried that the container will get into a state where it is not possible to build the file writer without wiping the CMake cache files. Hence the need to re-run Conan in order to re-generate the relevant supporting files.